releasenotes/notes/deprecate_arguments_and_deprecate_function-5e19f6f049fa489c.yaml

```
@@ -0,0 +1,7 @@
+---
+upgrade_misc:
+  - |
+    The ``deprecate_function`` and ``deprecate_arguments`` decorators had been deprecated since 0.24, released
+    on May 2023.
```

I would mention the removal explicitly:

```suggestion
    on May 2023, and have been removed in 2.0.
```